File: backend/dtos/auth_dto.py
from pydantic import BaseModel, EmailStr, Field, ConfigDict
from typing import Optional
from datetime import datetime


class LoginDTO(BaseModel):
    """DTO for user login request"""
    email: EmailStr
    password_hash: str = Field(..., min_length=6, description="User password")


class TokenResponseDTO(BaseModel):
    """DTO for authentication token response"""
    token: str
    token_type: str = "Bearer"


class UserResponseDTO(BaseModel):
    """DTO for user information response"""
    model_config = ConfigDict(from_attributes=True)
    
    id: str
    email: str
    created_at: datetime
    updated_at: datetime


class TokenValidationDTO(BaseModel):
    """DTO for token validation response"""
    valid: bool
    user: Optional[UserResponseDTO] = None

File: backend/src/dtos/auth.dto.ts
import { IsEmail, IsString, MinLength } from 'class-validator';

export class LoginDTO {
  @IsEmail({}, { message: 'Invalid email address' })
  email!: string;

  @IsString()
  @MinLength(6, { message: 'Password must be at least 6 characters' })
  password_hash!: string;
}

export class TokenResponseDTO {
  token!: string;
  token_type: string = 'Bearer';
}

export class UserResponseDTO {
  id!: string;
  email!: string;
  created_at!: Date;
  updated_at!: Date;
}

export class TokenValidationDTO {
  valid!: boolean;
  user?: UserResponseDTO;
}
